README.md

```diff
@@ -390,34 +386,40 @@ listened by some other service that stores the results in a database.
 
 ### Queue
 
-```ts
-Queue(queueName: string, redisPort: number, redisHost: string, redisOpts?: RedisOpts): Queue
-```
-```ts
-Queue(queueName: string, redisConnectionString: string, redisOpts? RedisOpts): Queue
+```typescript
+Queue(queueName: string, redisConnectionString?: string, opts: QueueOptions): Queue
 ```
 
 This is the Queue constructor. It creates a new Queue that is persisted in
 Redis. Everytime the same queue is instantiated it tries to process all the
 old jobs that may exist from a previous unfinished session.
 
+If no connection string or options passed, the queue will use ioredis default connection
+settings.
+
 __Arguments__
 
-```javascript
-    queueName {String} A unique name for this Queue.
-    redisPort {Number} A port where redis server is running.
-    redisHost {String} A host specified as IP or domain where redis is running.
-    redisOptions {Object} Options to pass to the redis client. https://github.com/luin/ioredis/blob/master/API.md#new-redisport-host-options 
+```typescript
+    queueName: string, // A unique name for this Queue.
+    redisConnectionString?: string, // string A connection string containing the redis server host, port and (optional) authentication.
+    opts?: QueueOptions, // Options to pass to the redis client. https://github.com/luin/ioredis/blob/master/API.md#new-redisport-host-options
 ```
 
-Alternatively, it's possible to pass a connection string to create a new queue.
-
-__Arguments__
-
-```javascript
-    queueName {String} A unique name for this Queue.
-    redisConnectionString {String} A connection string containing the redis server host, port and (optional) authentication.
-    redisOptions {Object} Options to pass to the redis client. https://github.com/luin/ioredis/blob/master/API.md#new-redisport-host-options
+```typescript
+  interface QueueOptions {
+    prefix?: string = 'bull',
+    redis : RedisOpts, // ioredis defaults
+    createClient?: (type: enum('client', 'subscriber'), redisOpts?: RedisOpts) => redisClient,
+
+    // Advanced settings
+    settings?: QueueSettings {
+      lockDuration?: number = 5000,
```

Comment:
What do you think about defaulting the lock to 30 seconds? My thoughts:
-A lot of people complain about jobs being double processed currently. So there must be a lot of poorly written job processor code out there that stalls the event loop :). Or other folks, like me, forget that we're running our code on tiny instances in the cloud where we forget that that it will max out the limited CPU. A 30sec timeout would give a bit more buffer. At least until we figure out a generic solution like #488.
-The tradeoff is that it will take longer for jobs to be considered 'stalled'. So instead waiting max 5sec to find out if a job was stalled, we'd wait 30sec. I think this is generally OK and that most people aren't running jobs that are that time-sensitive. Actual stalled jobs [due to process crashes] should be extremely rare anyways.

What do you think?